### PR TITLE
JIT ARM64_SVE: Add IF_SVE_BR_3A to IF_SVE_EX_3A

### PR DESCRIPTION
**Pull request by amanasifkhalid**
Part of #94549. Adds the following encodings:

* IF_SVE_BR_3A
* IF_SVE_BR_3B
* IF_SVE_BZ_3A
* IF_SVE_BZ_3A_A
* IF_SVE_CA_3A
* IF_SVE_EH_3A
* IF_SVE_EL_3A
* IF_SVE_EM_3A
* IF_SVE_EN_3A
* IF_SVE_EO_3A
* IF_SVE_EV_3A
* IF_SVE_EX_3A

cstool output:
```
trn1  z0.b, z1.b, z2.b
trn1  z3.h, z4.h, z5.h
trn2  z6.s, z7.s, z8.s
trn2  z9.d, z10.d, z11.d
uzp1  z12.b, z13.b, z14.b
uzp1  z15.h, z16.h, z17.h
uzp2  z18.s, z19.s, z20.s
uzp2  z21.d, z22.d, z23.d
zip1  z24.b, z25.b, z26.b
zip1  z27.h, z28.h, z29.h
zip2  z30.s, z31.s, z0.s
zip2  z1.d, z2.d, z3.d
trn1  z0.q, z1.q, z2.q
trn2  z3.q, z4.q, z5.q
uzp1  z6.q, z7.q, z8.q
uzp2  z9.q, z10.q, z11.q
zip1  z12.q, z13.q, z14.q
zip2  z15.q, z16.q, z17.q
tbl   z0.b, { z1.b }, z2.b
tbl   z3.h, { z4.h }, z5.h
tbx   z6.s, z7.s, z8.s
tbx   z9.d, z10.d, z11.d
tbl   z0.b, { z1.b, z2.b }, z2.b
tbl   z3.h, { z4.h, z5.h }, z5.h
tbl   z6.s, { z7.s, z8.s }, z8.s
tbl   z9.d, { z10.d, z11.d }, z11.d
tbxq  z0.b, z1.b, z2.b
tbxq  z3.h, z4.h, z5.h
tbxq  z6.s, z7.s, z8.s
tbxq  z9.d, z10.d, z11.d
sdot  z0.s, z1.b, z2.b
sdot  z3.d, z4.h, z5.h
udot  z6.s, z7.b, z8.b
udot  z9.d, z10.h, z11.h
smlalb        z0.h, z1.b, z2.b
smlalt        z3.s, z4.h, z5.h
smlslb        z6.d, z7.s, z8.s
smlslt        z9.h, z10.b, z11.b
umlalb        z12.s, z13.h, z14.h
umlalt        z15.d, z16.s, z17.s
umlslb        z18.h, z19.b, z20.b
umlslt        z21.s, z22.h, z23.h
sqrdmlah      z0.b, z1.b, z2.b
sqrdmlah      z3.h, z4.h, z5.h
sqrdmlsh      z6.s, z7.s, z8.s
sqrdmlsh      z9.d, z10.d, z11.d
sqdmlalbt     z0.h, z1.b, z2.b
sqdmlslbt     z3.s, z4.h, z5.h
sqdmlslbt     z6.d, z7.s, z8.s
sqdmlalb      z0.h, z1.b, z2.b
sqdmlalt      z3.s, z4.h, z5.h
sqdmlslb      z6.d, z7.s, z8.s
sqdmlslt      z9.h, z10.b, z11.b
sclamp        z0.b, z1.b, z2.b
sclamp        z3.h, z4.h, z5.h
uclamp        z6.s, z7.s, z8.s
uclamp        z9.d, z10.d, z11.d
tblq  z0.b, { z1.b }, z2.b
uzpq1 z3.h, z4.h, z5.h
uzpq2 z6.s, z7.s, z8.s
zipq1 z9.d, z10.d, z11.d
zipq2 z12.b, z13.b, z14.b
```

JitDisasm output:
```
trn1    z0.b, z1.b, z2.b
trn1    z3.h, z4.h, z5.h
trn2    z6.s, z7.s, z8.s
trn2    z9.d, z10.d, z11.d
uzp1    z12.b, z13.b, z14.b
uzp1    z15.h, z16.h, z17.h
uzp2    z18.s, z19.s, z20.s
uzp2    z21.d, z22.d, z23.d
zip1    z24.b, z25.b, z26.b
zip1    z27.h, z28.h, z29.h
zip2    z30.s, z31.s, z0.s
zip2    z1.d, z2.d, z3.d
trn1    z0.q, z1.q, z2.q
trn2    z3.q, z4.q, z5.q
uzp1    z6.q, z7.q, z8.q
uzp2    z9.q, z10.q, z11.q
zip1    z12.q, z13.q, z14.q
zip2    z15.q, z16.q, z17.q
tbl     z0.b, { z1.b }, z2.b
tbl     z3.h, { z4.h }, z5.h
tbx     z6.s, z7.s, z8.s
tbx     z9.d, z10.d, z11.d
tbl     z0.b, { z1.b, z2.b }, z2.b
tbl     z3.h, { z4.h, z5.h }, z5.h
tbl     z6.s, { z7.s, z8.s }, z8.s
tbl     z9.d, { z10.d, z11.d }, z11.d
tbxq    z0.b, z1.b, z2.b
tbxq    z3.h, z4.h, z5.h
tbxq    z6.s, z7.s, z8.s
tbxq    z9.d, z10.d, z11.d
sdot    z0.s, z1.b, z2.b
sdot    z3.d, z4.h, z5.h
udot    z6.s, z7.b, z8.b
udot    z9.d, z10.h, z11.h
smlalb  z0.h, z1.b, z2.b
smlalt  z3.s, z4.h, z5.h
smlslb  z6.d, z7.s, z8.s
smlslt  z9.h, z10.b, z11.b
umlalb  z12.s, z13.h, z14.h
umlalt  z15.d, z16.s, z17.s
umlslb  z18.h, z19.b, z20.b
umlslt  z21.s, z22.h, z23.h
sqrdmlah z0.b, z1.b, z2.b
sqrdmlah z3.h, z4.h, z5.h
sqrdmlsh z6.s, z7.s, z8.s
sqrdmlsh z9.d, z10.d, z11.d
sqdmlalbt z0.h, z1.b, z2.b
sqdmlslbt z3.s, z4.h, z5.h
sqdmlslbt z6.d, z7.s, z8.s
sqdmlalb z0.h, z1.b, z2.b
sqdmlalt z3.s, z4.h, z5.h
sqdmlslb z6.d, z7.s, z8.s
sqdmlslt z9.h, z10.b, z11.b
sclamp  z0.b, z1.b, z2.b
sclamp  z3.h, z4.h, z5.h
uclamp  z6.s, z7.s, z8.s
uclamp  z9.d, z10.d, z11.d
tblq    z0.b, { z1.b }, z2.b
uzpq1   z3.h, z4.h, z5.h
uzpq2   z6.s, z7.s, z8.s
zipq1   z9.d, z10.d, z11.d
zipq2   z12.b, z13.b, z14.b
```

cc @dotnet/arm64-contrib

### CODE REVIEW

**Comment by TIHan** [comment]
cc @a74nh

File: src/coreclr/jit/emitarm64.cpp [hunk 1]
```diff
@@ -22223,12 +22373,23 @@ BYTE* emitter::emitOutput_InstrSve(BYTE* dst, instrDesc* id)
...
+        case IF_SVE_BZ_3A:   // ........xx.mmmmm ......nnnnnddddd -- SVE table lookup (three sources)
+        case IF_SVE_BZ_3A_A: // ........xx.mmmmm ......nnnnnddddd -- SVE table lookup (three sources)
+        case IF_SVE_CA_3A:   // ........xx.mmmmm ......nnnnnddddd -- sve_int_perm_tbxquads
+        case IF_SVE_EH_3A:   // ........xx.mmmmm ......nnnnnddddd -- SVE integer dot product (unpredicated)
+        case IF_SVE_EL_3A:   // ........xx.mmmmm ......nnnnnddddd -- SVE2 integer multiply-add long
+        case IF_SVE_EM_3A:   // ........xx.mmmmm ......nnnnnddddd -- SVE2 saturating multiply-add high
+        case IF_SVE_EN_3A:   // ........xx.mmmmm ......nnnnnddddd -- SVE2 saturating multiply-add interleaved long
+        case IF_SVE_EO_3A:   // ........xx.mmmmm ......nnnnnddddd -- SVE2 saturating multiply-add long
+        case IF_SVE_EV_3A:   // ........xx.mmmmm ......nnnnnddddd -- SVE integer clamp
+        case IF_SVE_EX_3A:   // ........xx.mmmmm ......nnnnnddddd -- SVE permute vector elements (quadwords)
```
Comment (TIHan) [inline src/coreclr/jit/emitarm64.cpp:22392]: That's nice that you didn't have to add any additional encodings.

**Review by TIHan — APPROVED** [review]
LGTM!

File: src/coreclr/jit/emitarm64.cpp (hunk 1 above)
Comment (amanasifkhalid) [reply src/coreclr/jit/emitarm64.cpp:22392]: Yeah, these ones were pretty slick; the next batch should be similarly easy.

**Review by a74nh — APPROVED** [review]
LGTM. These are all following existing formats and styles.

**Pull request merged**